tests/integration/static_ip_address_test.py

```diff
@@ -918,6 +922,108 @@ def test_merge_ip_enabled_property_from_current(setup_eth1_static_ip):
     assertlib.assert_state_match(desired_state)
 
 
+@pytest.mark.skipif(
+    nm_minor_version() < 54,
+    reason="IPv4 forwarding is not supported on NM 1.54-",
+)
+def test_ipv4_sysctl_forwarding(setup_eth1_static_ip):
+    desired_state = {
+        Interface.KEY: [
+            {
+                Interface.NAME: "eth1",
+                Interface.IPV4: {
+                    InterfaceIPv4.ENABLED: True,
+                    InterfaceIPv4.ADDRESS: [
+                        {
+                            InterfaceIPv4.ADDRESS_IP: IPV4_ADDRESS1,
+                            InterfaceIPv4.ADDRESS_PREFIX_LENGTH: 24,
+                        }
+                    ],
+                    InterfaceIPv4.FORWARDING: True,
+                },
+            }
+        ]
+    }
+    apply_with_description(
+        "Enable IPv4 forwarding on eth1",
+        desired_state,
+    )
+    assertlib.assert_state_match(desired_state)
+    desired_state[Interface.KEY][0][Interface.IPV4][
+        InterfaceIPv4.FORWARDING
+    ] = False
+    apply_with_description(
+        "Disable IPv4 forwarding on eth1",
+        desired_state,
+    )
+    assertlib.assert_state_match(desired_state)
+
+
+@pytest.mark.skipif(
+    nm_minor_version() >= 54,
+    reason="IPv4 forwarding is supported on NM 1.54+",
+)
+def test_ipv4_sysctl_forwarding_unsupported_error(setup_eth1_static_ip):
+    cmdlib.exec_cmd(
+        "sysctl -w net.ipv4.conf.eth1.forwarding=1".split(), check=True
+    )
+    desired_state = {
+        Interface.KEY: [
+            {
+                Interface.NAME: "eth1",
+                Interface.IPV4: {
+                    InterfaceIPv4.ENABLED: True,
+                    InterfaceIPv4.ADDRESS: [
+                        {
+                            InterfaceIPv4.ADDRESS_IP: IPV4_ADDRESS1,
+                            InterfaceIPv4.ADDRESS_PREFIX_LENGTH: 24,
+                        }
+                    ],
+                    InterfaceIPv4.FORWARDING: False,
+                },
+            }
+        ]
+    }
+    with pytest.raises(libnmstate.error.NmstateVerificationError):
+        apply_with_description(
+            "Configure the ethernet device eth1 with IPv4 sysctl forwarding "
+            "enabled",
+            desired_state,
+        )
```

Comment:
I think we have to check that the verification error is due to the right reason. For that, maybe we can only check the content of the error message, which is a bit ugly but I think it's OK in this case. If we change the error message we'll just need to adapt a bit a single test.

Reply:
I changed the test to check the error message instead, thanks!